raft/tracker/inflights.go

```diff
@@ -22,37 +28,40 @@ package tracker
 type Inflights struct {
 	// the starting index in the buffer
 	start int
-	// number of inflights in the buffer
-	count int
 
-	// the size of the buffer
-	size int
+	count int    // number of inflight messages in the buffer
+	bytes uint64 // number of inflight bytes
+
+	size     int    // the max number of inflight messages
+	maxBytes uint64 // the max total byte size of inflight messages
```

Comment:
Point out that it's not a hard max - we might overshoot.  CRDB does it all the time as you know :-)
I would also guess the behavior here is to overshoot *every time* when we can. We'd never "not send" a message because that would put us over, right? Instead we make sure to overshoot only by one message. (I am guessing, since this is the beginning of my review).